### PR TITLE
Add Lectern Client to monorepo

### PR DESCRIPTION
**Pull request by joneubank**
This adds the existing Lectern Client code into this monorepo. Previously this was written and maintained in [overture-stack/js-lectern-client](https://github.com/overture-stack/js-lectern-client). This code will now be maintained in the same directory as the Lectern server so that types and utilities can be shared.

The code added in this request includes a mostly unchanged copy/paste of the lectern client in its most up to date form. A small list of changes were made to make it align with the processes of this repository, including:
- Updating to latest TS version
- Update tsconfig to match other packages in monorepo
- Formatting with prettier to update indents from spaces to tabs
- Cleaning up build scripts and removing old publishing notes
- Move build directory from `/lib` to `/dist`
- Updating copyright years

This accomplishes the base goals of: #206 , but does not touch the refactoring tasks such as updating the client to use the same types as the server.

### CODE REVIEW

File: libraries/dictionary/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	"main": "dist/index.js",
 	"scripts": {
 		"build": "pnpm build:clean && tsc",
-		"build:clean": "rm -rf dist/ && mkdir dist",
+		"build:clean": "rimraf -rf dist/ && mkdir dist",
```
Comment (leoraba) [inline libraries/dictionary/package.json:8]: no need arguments `-rf` when using rimraf

File: libraries/common/package.json [hunk 2]
```diff
@@ -5,9 +5,12 @@
 	"main": "dist/index.js",
 	"scripts": {
 		"build": "pnpm build:clean && tsc",
-		"build:clean": "rm -rf dist/ && mkdir dist"
+		"build:clean": "rimraf -rf dist/ && mkdir dist"
```
Comment (leoraba) [inline libraries/common/package.json:8]: no need arguments -rf when using rimraf

**Review by leoraba — APPROVED** [review]
just left a comment overall LGTM

File: packages/client/package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
+	"name": "@overturebio-stack/lectern-client",
```
Comment (joneubank) [inline packages/client/package.json:2]: Need to rename to `@overture-stack/lectern-client`. Once published, we should publish a deprecation notice with a link to the new package to the [original package location](https://www.npmjs.com/package/@overturebio-stack/lectern-client)

**Pull request merged**